api/models.go

```diff
@@ -69,3 +69,16 @@ type ChatSessionResponse struct {
 	UpdatedAt time.Time `json:"updatedAt"`
 	MaxLength int32     `json:"maxLength"`
 }
+
+type Pagination struct {
+	Page int32 `json:"page"`
+	Size int32 `json:"size"`
+	Data  []interface{} `json:"data"`
+	Total int64         `json:"total"`
+    
+}
+
+func (p *Pagination) Offset() int32 {
+	return (p.Page - 1) * p.Size
+}
+
```

Comment:
:robot: OpenAI

This patch adds a new `Pagination` struct and an `Offset()` method to calculate the offset. 

The `Pagination` struct looks good, but I have a few suggestions for the `Offset()` method:

- It would be helpful to add a comment explaining what this method does and how it is used.
- The method should return an error if `Size` is less than or equal to zero to avoid a potential division by zero error.

Here's an updated version of the `Offset()` method with these changes:

```go
func (p *Pagination) Offset() (int32, error) {
	if p.Size <= 0 {
		return 0, fmt.Errorf("invalid page size: %d", p.Size)
	}
	return (p.Page - 1) * p.Size, nil
}
```

Please let me know if you have any questions or concerns.

<!-- This is an auto-generated comment by OpenAI -->